### PR TITLE
CLOUDP-106123: [mongocli ] Implement mongocli auth logout

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ CLOUDP-106123

## Checklist
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments

```bash
./bin/mongocli auth logout --profile test
? Are you sure you want to log out No

./bin/mongocli auth logout --profile test
? Are you sure you want to log out Yes
Successfully logged out

./bin/mongocli auth logout --profile test
Error: not logged in
```

### CODE REVIEW

File: internal/cli/delete_opts.go [hunk 1]
```diff
@@ -54,6 +54,8 @@ func (opts *DeleteOpts) Delete(d interface{}, a ...string) error {
 
 	var err error
 	switch f := d.(type) {
+	case func() error:
```
Comment (gssbzn) [inline internal/cli/delete_opts.go:57]: adding support for a no "entry" case

File: internal/cli/auth/logout.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+	return opts.Delete(opts.config.Delete)
+}
+
+func LogoutBuilder() *cobra.Command {
+	opts := &logoutOpts{
+		DeleteOpts: cli.NewDeleteOpts("Successfully logged out\n", " "),
+	}
+
+	cmd := &cobra.Command{
```
Comment (blva) [inline internal/cli/auth/logout.go:69]: Do we need to add annotations to commands with no args? I've seen something like `		Annotations: map[string]string{}, ` for such scenario.

File: internal/cli/auth/logout.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	if _, err := opts.flow.Revoke(ctx, config.RefreshToken(), "refresh_token"); err != nil {
+		return err
+	}
+
+	return opts.Delete(opts.config.Delete)
+}
+
+func LogoutBuilder() *cobra.Command {
+	opts := &logoutOpts{
+		DeleteOpts: cli.NewDeleteOpts("Successfully logged out\n", " "),
```
Comment (blva) [inline internal/cli/auth/logout.go:66]: is the space intentional here in `" "`?

**Review by blva — COMMENTED** [review]
Minor comments

File: internal/cli/auth/logout.go (hunk 2 above)
Comment (gssbzn) [reply internal/cli/auth/logout.go:69]: I don't think we should there's no benefit to adding an empty map

File: internal/cli/auth/logout.go (hunk 3 above)
Comment (gssbzn) [reply internal/cli/auth/logout.go:66]: yes, if I don't there's a fallback message we don't want 

I was actually tempted to not use our deleter struct and just copy/paste what's needed so I'm happy to take that path

File: internal/cli/auth/logout.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			opts.OutWriter = cmd.OutOrStdout()
+			opts.config = config.Default()
+			return opts.initFlow()
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if config.RefreshToken() == "" {
+				return errors.New("not logged in")
+			}
+			if err := opts.PromptWithMessage("Are you sure you want to log out"); err != nil {
```
[the remark this answers is not in the document]
Comment (gssbzn) [reply internal/cli/auth/logout.go]: added

**Review by blva — APPROVED** [review]

**Pull request merged**